src/OrchardCore/OrchardCore.FileStorage.Abstractions/IFileStore.cs

```diff
@@ -39,12 +39,13 @@ public interface IFileStore
         /// Enumerates the content (files and directories) in a given directory within the file store.
         /// </summary>
         /// <param name="path">The path of the directory to enumerate, or <c>null</c> to enumerate the root of the file store.</param>
+        /// <param name="includeSubDirectories">A flag to indicate whether to get the contents from just the top directory or from all sub-directories as well.</param>
         /// <returns>The list of files and directories in the given directory.</returns>
         /// <exception cref="FileStoreException">Thrown if the specified directory does not exist.</exception>
         /// <remarks>
         /// Results are grouped by entry type, where directories are followed by files.
         /// </remarks>
-        Task<IEnumerable<IFileStoreEntry>> GetDirectoryContentAsync(string path = null);
+        Task<IEnumerable<IFileStoreEntry>> GetDirectoryContentAsync(string path = null, bool includeSubDirectories = false);
```

Comment:
I am not a fan of doing this, but it might help improve perf in some scenarios, so I will be ok with it.